### PR TITLE
Proposal model and documentation

### PR DESCRIPTION
**Pull request by nitrosx**
## Description

This PR add better documentation for the Proposal model and endpoint. It includes improvements to the swagger explorer UI for this endpoint.
A complete review of the model has been also done, together with renaming the class model to ProposalClass

## Motivation

Improving documentation, naming, and swagger integration.
Review model, eliminated inconsistencies, and ported field constrains from old backend

## Changes:

* Proposal model class and dto
* Swagger decorators for proposals mode, dto and controller

## Tests included/Docs Updated?

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?

### CODE REVIEW

File: src/common/interfaces/common.interface.ts [hunk 1]
```diff
@@ -40,6 +41,12 @@ export interface ILimitsFilter {
   order: string;
 }
 
+export class ILimitsFilterDto implements ILimitsFilter {
```
Comment (martin-trajanovski) [inline src/common/interfaces/common.interface.ts:44]: As I see this ILimitsFilterDto is completly the same as ILimitsFilter. Do we really need two different representations or we can have just one?

File: src/common/interfaces/common.interface.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//import { ApiProperty } from "@nestjs/swagger";
```
Comment (martin-trajanovski) [inline src/common/interfaces/common.interface.ts:1]: This comment can be removed.

File: src/common/utils.ts [hunk 3]
```diff
@@ -623,3 +623,40 @@ export const addUpdatedFields = <T>(
...
+export const fullQueryDescription = '<pre>\n \
+{\n \
+  "fields?": [ "field1", "field2", ...],\n \
+  "limits?": {\n \
+    "limit": number,\n \
+    "skip": number,\n \
+    "order": [ascending, descending]\n \
+  }\n \
+}\n \
+</pre>';;
```
Comment (martin-trajanovski) [inline src/common/utils.ts:662]: All these descriptions are a bit confusing but I understand the point it is good to have better documentations for the filters and queries.

File: src/proposals/proposals.controller.ts [hunk 4]
```diff
@@ -97,26 +114,51 @@ export class ProposalsController {
...
   @Get()
+  //@ApiExtraModels(IProposalFilterDto)
   @ApiQuery({
     name: "filters",
-    description: "Database filters to apply when retrieve all proposals",
+    description: "Database filters to apply when retrieve proposals\n" + filterDescription,
     required: false,
+    //schema: {
+    //  $ref: getSchemaPath(IProposalFilterDto),
+    //}
```
Comment (martin-trajanovski) [inline src/proposals/proposals.controller.ts:126]: Comments can be removed here

File: src/proposals/proposals.controller.ts [hunk 5]
```diff
@@ -215,49 +344,86 @@ export class ProposalsController {
...
     return this.attachmentsService.findOneAndRemove({
       _id: attachmentId,
       proposalId: proposalId,
     });
   }
 
-  // POST /proposals/:id/datasets
-  @UseGuards(PoliciesGuard)
+  // POST /proposals/:pid/datasets
+/*   @UseGuards(PoliciesGuard)
```
Comment (martin-trajanovski) [inline src/proposals/proposals.controller.ts:378]: I see you have commented this part and if it is not needed anymore maybe removing it is better.

File: src/proposals/proposals.controller.ts [hunk 6]
```diff
@@ -215,49 +344,86 @@ export class ProposalsController {
...
+  })
   async findAllDatasets(
-    @Param("id") proposalId: string,
+    @Param("pid") proposalId: string,
   ): Promise<DatasetClass[] | null> {
     return this.datasetsService.findAll({ where: { proposalId } });
   }
 
-  // PATCH /proposals/:id/datasets/:fk
+/*   // PATCH /proposals/:id/datasets/:fk
```
Comment (martin-trajanovski) [inline src/proposals/proposals.controller.ts:426]: I see this part is also commented. Are we using it or it should be removed.

**Review by martin-trajanovski — CHANGES_REQUESTED** [review]

File: src/common/interfaces/common.interface.ts (hunk 1 above)
Comment (nitrosx) [reply src/common/interfaces/common.interface.ts:44]: You are correct!!!
I'm going to remove it.

File: src/common/utils.ts (hunk 3 above)
Comment (nitrosx) [reply src/common/utils.ts:662]: I agree with you!!!
I want to get these info in, although I'm absolutely convinced that this is not the best and most efficient way.

File: src/proposals/proposals.controller.ts (hunk 5 above)
Comment (nitrosx) [reply src/proposals/proposals.controller.ts:378]: Done. We want to provide only the functionality to return the linked datasets, but not to create/update datasets from proposals

File: src/proposals/proposals.controller.ts (hunk 6 above)
Comment (nitrosx) [reply src/proposals/proposals.controller.ts:426]: Done!!!

**Pull request merged**